### ISSUE TITLE
Bug: False positive `Match must be exhaustive [E0004]`

### ISSUE THREAD

**Issue by NixyJuppie** [intellij-rust/intellij-rust#10759]
## Environment

* **IntelliJ Rust plugin version:** 0.4.199.5415-232
* **Rust toolchain version:** 1.71.0 (8ede3aae2 2023-07-12) x86_64-unknown-linux-gnu
* **IDE name and version:** JetBrains Rider 2023.2 RC 1 (RD-232.8660.194)
* **Operating system:** Linux 6.3.10-203.fsync.fc38.x86_64
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr

## Problem description
Plugin reports error `Match must be exhaustive [E0004]` for exhaustive matches.
It works when matching with consts `MIN` and `MAX`, but does not work with literals.

## Steps to reproduce
```rs
fn example(value: u16) {
    // No error here
    match value {
        u16::MIN..=u16::MAX => {}
    }

    // Match must be exhaustive [E0004]
    match value {
        0..=65535 => {}
    }
}
```

**Comment by Maksim20023** [comment]
Hi! Thank you for the report!
This is a known issue for us and has already been reported: see #4311 . When we have any new information about it, we will add a reply to this ticket. You can subscribe to it to stay informed and vote (click the emoji button in the bottom-left corner of the GitHub post). If you have any further suggestions, please feel free to reach out.
